### DISCUSSION TITLE
CmdToggleBookmarks works differently with F12 and when shortcut assigned

### DISCUSSION THREAD

**Discussion by selfmadebuba** [sumatrapdfreader/sumatrapdf#2831] [Q&A]
CmdToggleBookmarks when CUSTOM shortcut is assigned and used **works incorrectly**:
* opens and focuses bookmark window (it's possible to use cursor UP/DOWN on bookmark list)
* after another F12 it **DOES NOT UNFOCUS** bookmark window and it's **NOT CLOSING** bookmarks properly
(basically it's not a toggle in this case)

CmdToggleBookmarks when using default F12: (**here it works properly**)
* opens and focuses bookmark window (it's possible to use cursor UP/DOWN on bookmark list)
* on subsequent F12 it **UNFOCUSES** bookmark window and close it properly
(it's a toggle in this case)

**Comment by kjk** [comment]
It's unfortunate but there's no easy fix.

To make shortcuts customizable I had to use accelerators.

However, accelerators are processed before any processing in controls which breaks  controls like edit control or tree view (used for bookmarks) because a key used for a shortcut is necessary to implement control behavior but it'll be processed as accelerator and never seen by the control.

I added hacks to make many shortcut keys be processed as accelerators in tree view (F12 is one of them) but you must have assigned a key that is not processed as accelerator in tree view control (used in bookmarks view).
